src/Features/Core/Portable/LanguageServices/SymbolDisplayService/AbstractSymbolDisplayService.AbstractSymbolDescriptionBuilder.cs

```diff
@@ -687,6 +689,17 @@ private IEnumerable<SymbolDisplayPart> TypeParameterName(string text)
             {
                 return Part(SymbolDisplayPartKind.TypeParameterName, text);
             }
+
+            protected static async Task<List<SymbolDisplayPart>> GetClassifiedSymbolDisplayPartsAsync(
+                SemanticModel semanticModel, TextSpan textSpan, Workspace workspace,
+                bool insertSourceTextInGaps = false,
+                CancellationToken cancellationToken = default(CancellationToken))
+            {
+                var classifiedSpans = Classifier.GetClassifiedSpans(semanticModel, textSpan, workspace, cancellationToken);
+                var sourceText = await semanticModel.SyntaxTree.GetTextAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
interesting API.  I'm somewhat surprised it operates on a semantic model instead of, say, a Document.